presto-main/src/main/java/com/facebook/presto/operator/aggregation/AccumulatorCompiler.java

```diff
@@ -130,7 +131,9 @@ public static GenericAccumulatorFactoryBinder generateAccumulatorFactoryBinder(A
 
         // Generate methods
         generateAddInput(definition, stateField, inputChannelsField, maskChannelField, metadata.getInputMetadata(), metadata.getInputFunction(), callSiteBinder, grouped);
-        generateAddInputWindowIndex(definition, stateField, metadata.getInputMetadata(), metadata.getInputFunction(), callSiteBinder);
+        generateAddOrRemoveInputWindowIndex(definition, stateField, metadata.getInputMetadata(), metadata.getInputFunction(), "addInput", callSiteBinder);
+        metadata.getRemoveInputFunction().ifPresent(
```

Comment:
I think in case of not having remove input function, we still should generate a function matching the implemented interface but with, let's say the only statement in the body to be `throw new IllegalStateException`.
However, I didn't find anything to support the statement that your implementation may be illegal/may be prone to the undefined behavior, based on https://docs.oracle.com/javase/specs/jvms/se8/html/jvms-4.html and https://docs.oracle.com/javase/specs/jvms/se8/html/jvms-6.html#jvms-6.5.invokeinterface
But I think a good rule of thumb is not to generate the code which cannot be generated straightforwardly by the compiler.
So, let's treat this comment as a note/warning, not necessarily an "error"/must change.